### ISSUE TITLE
Screen reader announcements about highlighted text jarring and awkward

### ISSUE THREAD

**Issue by mattdricker** [hypothesis/product-backlog#1423]
This feedback comes out of Accessibility Web Audit from UC Berkeley.

There were a couple of specific comments in the above review about this:

> Hypothesis alerts to the screen reader when highlighted text begins and ends, but this made the flow jarring to hear.
>
>Is it possible to add a pause? It sounded like a part of the sentence was missing. In markup, try putting a period or some other way to indicate a break.

And the following comment was made in a separate section of the review:

> Highlight start and end were announced for every block-level element within a single highlight - this was too verbose and made it impossible to understand where one highlight started and another ended.
>
>Do not apply the highlight-start class to every element within the highlight. Instead, provide an announcement at the beginning and end of the highlight

We have also received feedback from another user frustrated that when the highlighted text begins in the middle of a word, the screen reader will only read the partial word prior to the highlighted text, then announces the start of the highlight, then continues reading the rest of the interrupted word. This completely breaks the flow of the text.

I see this primarily as an ergonomic issue about announcing highlights with the text, and as such there is not likely to be one single answer or fix.

Possible mitigation of this could include:

- Affordances for the user to dial down/up the verbosity of highlighted text announcements (in addition to the existing option to turn on/off highlights entirely)
- Announcing at beginning of parent element that x number of annotations/highlights are within subsequent section
- Navigation controls to jump to highlighted sections
- Recognizing when highlight begins/ends in the middle of word, and allowing adjustment of announcement accordingly
